core/src/main/java/eu/mhutti1/utils/storage/StorageDeviceUtils.kt

```diff
@@ -63,9 +63,9 @@ object StorageDeviceUtils {
       ?.map { dir -> StorageDevice(dir, false) }
       .orEmpty()
 
-  private fun environmentDevices() =
+  private fun environmentDevices(context: Context) =
     StorageDevice(
-      generalisePath(Environment.getExternalStorageDirectory().path, false),
+      generalisePath(context.getExternalFilesDir("").toString(), true),
       Environment.isExternalStorageEmulated()
```

Comment:
This `environmentDevices` function is responsible for getting Environment external storage directory. Which can be shared by other applications.  I don't understand why do we wanna switch to `getExternalFilesDir`which is internal to the application and it's not visible to other applications.

Reply:
Hi gouri-panda,

Actually ```Environment.getExternalStorageDirectory().path ``` is not working on android 10 and above because this method is depreciated. On android 10 and above its not returning any data. 

you can check the link.
https://developer.android.com/reference/android/os/Environment#getExternalStorageDirectory()

Reply:
Sure! but is the alternative API is the right approach? There are a couple of things that don't satisfy our requirements.
1. It will be our private internal folder. Other applications can't search the zim files if they want to.
2. The data inside the folder will be deleted if the user wants to uninstall it.

Is `MediaStore` API will be the right option?

Reply:
hi @gouri-panda ,

1. Yes according to current flow all zim files saved in app specific data directory ,app specific data directory can be shareable to other app  only below android 10. From android 10 and above they restricted access to another app data directory so we must need to move to public folders.
2. By changing using the ```Mediastore Api``` we can fix some of the existing issue like https://github.com/kiwix/kiwix-android/issues/2690.

So i think ```Mediastore Api``` will be better one.

Reply:
hi @gouri-panda and @kelson42 ,

I figure it out the solutions that app data must be in shareable folder with both internal and external storage. this will resolve lots of existing issues.

### Problem statement 
I tried with both ```file storage``` and ```MediaStore API``` but the problem is detection of  sdcard as secondary storage according to google doc on sdcard we can only create app specific folder. 
``` 
The WRITE_EXTERNAL_STORAGE permission must only grant write access to the primary external 
storage on a device. Apps must not be allowed to write to secondary external storage 
devices, except in their package-specific directories as allowed by synthesized 
permissions. Restricting writes in this way ensures the system can clean up files
when applications are uninstalled.
```

## Internal storage
a) I am able to create folder at topmost directory as kiwix and able to save zim file.

## External Storage
a) Since its detecting the sd card as secondary storage and not allowing to create public directory programmatically. App specific directory creation is working but our problem statement were different so only way to achieve this by issue the intent for ```ACTION_OPEN_DOCUMENT_TREE```. It will open file manager to select folder to save zim files on external storage.

I am following this solutions and its working .

https://stackoverflow.com/questions/35996341/android-gain-priviledges-to-write-to-secondary-sd-card-at-runtime